Mathlib/RingTheory/Nilpotent/Exp.lean

```diff
@@ -165,3 +167,87 @@ theorem exp_of_nilpotent_is_unit {a : A} (h : IsNilpotent a) : IsUnit (exp a) :=
   rw [← exp_add_of_commute h₁.symm h₂ h, neg_add_cancel a, exp_zero_eq_one]
 
 end IsNilpotent
+
+namespace LinearMap
+
+variable {R M N : Type*} [CommRing R]
+  [AddCommGroup M] [Module R M] [AddCommGroup N] [Module R N]
+
+variable (M) in
+theorem isNilpotent_lTensor_of_isNilpotent (f : Module.End R N) (hf : IsNilpotent f) :
+    IsNilpotent (f.lTensor M) := by
+  obtain ⟨k, hk⟩ := hf
+  exact ⟨k, by ext; simp [hk]⟩
+
+variable (N) in
+theorem isNilpotent_rTensor_of_isNilpotent (f : Module.End R M) (hf : IsNilpotent f) :
+    IsNilpotent (f.rTensor N) := by
+  obtain ⟨k, hk⟩ := hf
+  exact ⟨k, by ext; simp [hk]⟩
+
+open IsNilpotent TensorProduct
+
+variable [Module ℚ M]
+
+variable (N) in
+theorem rTensor_exp (f : Module.End R M) (hf : IsNilpotent f) :
+    exp (f.rTensor N) = (exp f).rTensor N := by
+  obtain ⟨k, hk⟩ := isNilpotent_rTensor_of_isNilpotent N f hf
```

Comment:
I think this follows more generally from a lemma matching `NormedSpace.exp_smul `?